data/mp/multiplay/script/rules/endconditions.js

```diff
@@ -1,62 +1,363 @@
-// /////////////////////////////////////////////////////////////////
-// END CONDITIONS
-function checkEndConditions()
+// registrate events conditions_eventGameInit, conditions_eventDroidBuilt, conditions_eventStructureBuil, conditions_eventResearched, conditions_eventAttacked
+namespace("conditions_");
+
+const STATE_contender = "contender";
+const STATE_winner = "winner";
+const STATE_loser = "loser";
+
+const STRUCTS = [FACTORY, CYBORG_FACTORY, VTOL_FACTORY]; // structures in which you can continue to play
+// TODO remove HQ. The destruction of the HQ will hide the minimap.
+
+
+// The time that the player's inactivity is allowed. Actions are considered
+// - unit building
+// - completion of the research
+// - construction of base structures (factories, power plants, laboratories, modules and oil rigs)
+// - dealing damage
+const IDLETIME = 5 * 60 * 1000;
+const BASESTRUCTS = {[FACTORY]: true, [CYBORG_FACTORY]: true, [VTOL_FACTORY]: true, [HQ]: true, [RESOURCE_EXTRACTOR]: true, [POWER_GEN]: true, [RESEARCH_LAB]: true};
+
+const SPOTTER = {
+	x: mapWidth / 2,
+	y: mapHeight / 2,
+	radius: (Math.sqrt(mapWidth * mapWidth + mapHeight * mapHeight) / 2) * 128,
+};
+
+var teams = []; // array class instance Team
+var playersTeam = new Array(maxPlayers); // array class instancePlayer
```

Comment:
It will be necessary to set these somewhere (maybe inside eventGameInit) away from global scope for saves to function properly.

eventGameLoaded is called when loading saves if you think it is necessary.

Reply:
Moved